trappy/plotter/ILinePlotGen.py

```diff
@@ -189,28 +207,28 @@ def _check_add_scatter(self, fig_params):
 
         fig_params["pointSize"] = self._attr["point_size"]
 
-    def add_plot(self, plot_num, data_dict, title="", test=False):
+    def add_plot(self, plot_num, data_frame, title="", test=False):
         """Add a plot for the corresponding index
 
         :param plot_num: The linear index of the plot
         :type plot_num: int
 
-        :param data_dict: The data for the plot
-        :type data_dict: dict
+        :param data_frame: The data for the plot
+        :type data_frame: :mod:`pandas.DataFrame`
 
         :param title: The title for the plot
         :type title: str
         """
 
-        datapoints = sum(len(v) for _, v in data_dict.iteritems())
+        datapoints = sum(len(v) for _, v in data_frame.iteritems())
```

Comment:
Since we are improving performance, here is another small improvement (7 times faster :smile:):

```python
datapoints = len(data_frame) * len(data_frame.columns)
```